scripts/network.js

```diff
@@ -249,100 +301,233 @@ export class ExplorerNetwork extends Network {
-                        )
-                        .map((tx) => {
-                            return {
-                                id: tx.txid,
-                                time: tx.blockTime,
-                                blockHeight: tx.blockHeight,
-                                amount: (txSum(tx.vout) - txSum(tx.vin)) / COIN,
-                            };
-                        })
-                        .filter((tx) => tx.amount != 0)
+
+            // Process our aggregated history data
+            if (
+                (cData && cData.transactions) ||
+                (cRecentTXs && cRecentTXs.transactions)
+            ) {
+                // Process Older (historical) TXs
+                const arrOlderTXs = this.toHistoricalTXs(
+                    cData.transactions || [],
+                    mapPaths
                 );
 
-                // If the results don't match the full 'max/requested results', then we know the rewards are complete
-                if (cData.transactions.length !== cData.itemsOnPage) {
-                    this.areRewardsComplete = true;
+                // Process Recent TXs, then add them manually on the basis that they are NOT already known in history
+                const arrRecentTXs = this.toHistoricalTXs(
+                    cRecentTXs.transactions || [],
+                    mapPaths
+                );
+                for (const cTx of arrRecentTXs) {
+                    if (
+                        !this.arrTxHistory.find((a) => a.id === cTx.id) &&
+                        !arrOlderTXs.find((a) => a.id === cTx.id)
+                    ) {
+                        // No identical Tx, so prepend it!
+                        this.arrTxHistory.unshift(cTx);
+                    }
+                }
+                this.arrTxHistory = this.arrTxHistory.concat(arrOlderTXs);
+
+                // If the results don't match the full 'max/requested results', then we know the history is complete
+                if (
+                    cData.transactions &&
+                    cData.transactions.length !== cData.itemsOnPage
+                ) {
+                    this.isHistorySynced = true;
                 }
             }
-            return this.arrRewards;
+            return this.arrTxHistory;
         } catch (e) {
             console.error(e);
         } finally {
-            this.rewardsSyncing = false;
+            this.historySyncing = false;
         }
     }
 
+    /**
+     * Convert a list of Blockbook transactions to HistoricalTxs
+     * @param {Array<object>} arrTXs - An array of the Blockbook TXs
+     * @param {Map} mapPaths - A map of the derivation paths for involved addresses
+     * @returns {Array<HistoricalTx>}
+     */
+    toHistoricalTXs(arrTXs, mapPaths) {
+        const txSum = (v) =>
+            v.reduce(
+                (t, s) =>
+                    t +
+                    (s.addresses &&
+                    s.addresses
```

Comment:
```suggestion
                     s.addresses.some((strAddr) => mapPaths.get(strAddr))
```

Reply:
This suggestion does not run, it's trying to run `.map` on the output of `.some` (boolean), I'm not sure what was intended here.

Reply:
Ahhhh, I see, good nit; save some CPU cycles!

Also, multi-line suggestions are done by "dragging" the cursor along the lines you wish to suggest on (the line numbers in the left side of the GitHub review editor/mode).